test/unit/Object/ConstantDirectoryTest.php

```diff
@@ -26,10 +28,7 @@ public function test_it_is_initialisable()
 
     public function test_it_returns_empty_array_with_no_constants()
     {
-        $this->assertSame(
-            [],
-            $this->newSubject(NoConstantsClass::class)->listConstants()
-        );
+        $this->assertCount(0, $this->newSubject(NoConstantsClass::class)->listConstants());
```

Comment:
I'd prefer to be explicit that you get an empty array (as the test name says) rather than any kind of empty countable / traversable instance.

Reply:
I will revert this assertion :+1: .